scripts/stack_usage/stack_usage.py

```diff
@@ -0,0 +1,671 @@
+    print('\t' + 'A function may be called within an inlined function but because the parent function was inlined, the function call will be linked to the root function who inlined the parent function.')
+    print('\t' + 'As such, the depth of the function call will be under the root function, not its parent function in source code.' + '\n')
+    print('\t' + 'If there are no function calls under a file, it means that the file does not contain any syscalls.' + '\n')
+    print('\t' + 'Memory usage is in bytes. If a function has multiple instances across several object files, the memory usage will be the maximum usage found across all instances.' + '\n')
+    print('Informational flags:' + '\n')
+    print('\t' + 'inlined: Function is inlined')
+    print('\t' + 'pointer: Function is called through a function pointer')
+    print('\t' + 'external: Function is not defined in the same object file')
+    print('\t' + 'static/dynamic/bounded: Flags determined by the .su file generated with -fstack-usage flag when CONFIG_STACK_USAGE is enabled.')
+    print('\t' + '\t' + '\t' + '\t' + 'See https://gcc.gnu.org/onlinedocs/gnat_ugn/Static-Stack-Usage-Analysis.html for more information.')
+    print('\t' + '*: Function was not found in any .su file and has no data on maximum stack usage' + '\n')
+
+
+# This is the main function of the script.
+# It calls all the other functions to generate the output
+# and checks for input flags and arguments when the
+# script is called.
+
+def main():
+
+    verbose = False
+
+    out_file = ''
+
+    len_args = len(sys.argv)
+
+    for i in range(1, len_args):
+
+        if sys.argv[i] == '-h':
+            print_help()
+            return
+
+        elif sys.argv[i] == '-v':
+            verbose = True
+
+        elif '-' not in sys.argv[i]:
+            out_file = sys.argv[i]
+            if not out_file.endswith('.txt'):
+                print('Output file must be a .txt file')
+                print_help()
+                return
+            if '.txt' in out_file and len(out_file) == 4:
+                print('Please specify output file name')
+                print_help()
+                return
+            if '/' not in out_file:
+                print('Path not specified, output file will be saved in the current directory')
+
+    if out_file == '':
+        print('Please specify a .txt output file')
+        print_help()
+        return
+
+    # flags structure:
+    # Key = function address
+    # Value = [flag1, flag2, ...]
+    # Note: values can be 'inlined', 'pointer', 'external', 'dyanamic,bounded', 'static' or '*'
+    flags = {}
+
+    # cuOffsets structure:
+    # Key = file name
+    # Value = compilation unit offset for that file
+    cuOffsets = {}
+
+    max_usage = {}
+
+    mem_by_name = {}
+
+    stack_size = [0]
+
+    # If ELF file is located elsewhere, please change the path here.
+    info = getElf('build/zephyr/zephyr.elf')
```

Comment:
I think it would be better to specify the ELF as an argument to the script, rather than assume its location.